Update instructions on how to update constraints files (#672)
napari/docs: docs/developers/coredev/release.md

To get updated constraints for a PR, use `@napari-bot update constraints` in a PR comment, then follow the instruction added by the bot to the conversation.

````{admonition} Example

> To manually update the constraint files, you need to have the [`uv`](https://github.com/astral-sh/uv) and copy 

Reviewer: ```suggestion
To manually update the constraints files, you need to have the [`uv`](https://github.com/astral-sh/uv) tool installed, then copy 
```

> the command from the second line of a given constraints file. 

Reviewer: ```suggestion
and run the command from the second line of a given constraints file. 
```

the command from the second line of a given constraints file. 
For example:  

```bash

> uv pip compile --python-version 3.11 --upgrade --output-file resources/constraints/constraints_py3.11_docs.txt pyproject.toml resources/constraints/version_denylist.txt resources/constraints/version_denylist_examples.txt ../docs/requirements.txt resources/constraints/pydantic_le_2.txt --extra pyqt5 --extra pyqt6 --extra pyside2 --extra pyside6_experimental --extra testing --extra testing_extra --extra optional

Reviewer: Shouldn't `resources/constraints/version_denylist_examples.txt` be kept (as in https://github.com/napari/napari/pull/7788 ) ?

Author: I have changed from docs to normal constraints, as docs are still on python 3.10. 

Once we understand the source of problems in #7714 we may get docs constraints here.

Reviewer 2: I think this is now resolved as napari/napari#7714 is merged! Can we add this in here?

Reviewer 3: @melissawm I think bumping to 3.12 makes sense now that #7714 has landed.

> uv pip compile --python-version 3.12 --output-file resources/constraints/constraints_py3.12.txt pyproject.toml resources/constraints/version_denylist.txt --extra pyqt6 --extra pyside2 --extra pyside6_experimental --extra testing --extra testing_extra --extra optional

Reviewer 4: this updates 3.12, we want to update docs though, right? I think this does that

```suggestion
uv pip compile --python-version 3.10 --output-file resources/constraints/constraints_py3.10_docs.txt pyproject.toml resources/constraints/version_denylist.txt --extra pyqt6 --extra pyside2 --extra pyside6_experimental --extra testing --extra testing_extra --extra optional
```

Reviewer 2: We can now use 3.12 here 🎉
